src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs

```diff
@@ -291,33 +271,59 @@ public static Quaternion Add(Quaternion value1, Quaternion value2)
         /// <returns>A new quaternion representing the concatenation of the <paramref name="value1" /> rotation followed by the <paramref name="value2" /> rotation.</returns>
         public static Quaternion Concatenate(Quaternion value1, Quaternion value2)
         {
-            Quaternion ans;
+            if (Vector128.IsHardwareAccelerated)
+            {
+                var left = Unsafe.BitCast<Quaternion, Vector128<float>>(value1);
+                var right = Unsafe.BitCast<Quaternion, Vector128<float>>(value2);
 
-            // Concatenate rotation is actually q2 * q1 instead of q1 * q2.
-            // So that's why value2 goes q1 and value1 goes q2.
-            float q1x = value2.X;
-            float q1y = value2.Y;
-            float q1z = value2.Z;
-            float q1w = value2.W;
+                var c = Cross(right, left);
+                var dot = Vector3.Dot(right.AsVector3(), left.AsVector3());
 
-            float q2x = value1.X;
-            float q2y = value1.Y;
-            float q2z = value1.Z;
-            float q2w = value1.W;
+                var t0 = right * Vector128.Shuffle(left, Vector128.Create(3, 3, 3, 3));
+                var t1 = left * Vector128.Shuffle(right, Vector128.Create(3, 3, 3, 3)) + c;
+                var t2 = t0 + t1;
+                var ans = Unsafe.BitCast<Vector128<float>, Quaternion>(t2);
+                ans.W = t0.GetElement(3) - dot;
+                return ans;
 
-            // cross(av, bv)
-            float cx = q1y * q2z - q1z * q2y;
-            float cy = q1z * q2x - q1x * q2z;
-            float cz = q1x * q2y - q1y * q2x;
+                static Vector128<float> Cross(Vector128<float> l, Vector128<float> r)
+                {
+                    return (Vector128.Shuffle(l, Vector128.Create(1, 2, 0, 3)) *
+                            Vector128.Shuffle(r, Vector128.Create(2, 0, 1, 3))) -
+                           (Vector128.Shuffle(l, Vector128.Create(2, 0, 1, 3)) *
+                            Vector128.Shuffle(r, Vector128.Create(1, 2, 0, 3)));
+                }
+            }
```

Comment:
This implementation is notably quite different from the one that `operator *` does. `operator *` generally follows the algorithm as seen in places like: https://github.com/microsoft/DirectXMath/blob/main/Inc/DirectXMathMisc.inl#L82

Reply:
This is left over from me benchmarking different multiplication methods. Will be cleaned up by replacing the entire method with =>value2 * value1